bindings/dotnet/openDAQ.Net/README.md

```diff
@@ -0,0 +1,201 @@
+
+## Table of Contents
+- [Prerequisites](#prerequisites)  
+- [C# to C++ access considerations](#c-to-c-access-considerations)  
+   - [Reference counting](#reference-counting)  
+   - [Memory consumption](#memory-consumption)  
+- [Bindings](#bindings)  
+   - [Auto generated code](#auto-generated-code)  
+   - [_openDAQ_ core-type conversion](#opendaq-core-type-conversion)  
+   - [Handling new interfaces](#handling-new-interfaces)  
+   - [Inline documentation](#inline-documentation)  
+   - [Special implementations](#special-implementations)  
+
+
+## Development environment
+- Visual Studio 2022 (V17.8.x)  
+- TargetFramework: net8.0  
+- Platform: x64  
+- Language: C#  
+- Test-framework: NUnit  
+
+## Prerequisites
+The _openDAQ_ binaries (on Windows `*.dll`) need to exist in `./build/x64/msvc-22/full/bin/Debug`
+or `*/Release`. This is normally accomplished by running `cmake`.  
+The libraries will be copied to the output directory .  
+For `NuGet` the libraries will be placed in the structure `runtime/win-x64/native/*`.  
+The generated C# code (`RTGen`) needs to reside in `./build/bindings/CSharp/core` (also from
+`cmake`).  
+The manually created C# code resides in `./bindings/dotnet/openDAQ.Net/openDAQ.Net/core`.
+
+```
+./build
+  /bindings/CSharp/core
+    /coreobjects/*
+    /opendaq/*
+    /*DllInfo.cs
+  /x64/msvc-22/full/bin
+    /Debug
+      /*-debug.dll
+      /*-debug.module.dll
+    /Release
+      /*.dll
+      /*.module.dll
+./openDAQ.Net/openDAQ.Net
+  /core/*
+  /...
+```
+
+## C# to C++ access considerations
+
+### Reference counting
+The _openDAQ_ SDK is written in C++ and all objects implement the `IUnknown` interface
+(interface querying and reference counting). In C++ (and Python) _smart pointers_ are used
+which automatically handle _object destruction_ and thus _reference counters_.  
+
+When used in C#, only object pointers (`IntPtr`) will be exchanged. Dereferencing a C++
+object has to be handled explicitly (the `BaseObject` implements the _Disposable_ pattern,
+which dereferences the underlying C++ object).  
+Though usually a .NET developer rarely would use `obj.Dispose()` directly, it will be
+automatically called on (.NET) object finalization / destruction by the
+_.NET Garbage Collector_.  
+
+### Memory consumption
+Internally we agreed that it will not be necessary to always dispose of those C# objects,
+because eventually the _.NET Garbage Collector_ will collect (and finalize/destroy)
+unreferenced (.NET) objects, which in turn dereferences the C++ object in the
+finalizer of `BaseObject` by calling `this.Dispose()`.  
+Regarding memory consumption, this is similar to collecting pure .NET objects that are no longer
+referenced. The user has still the choice to manually free up no longer needed objects by
+calling the objects `Dispose()` method (or by using the `using` statement to automatically dispose
+the object when leaving the scope).  
```

Comment:
I'd omit things like "Internally we agreed" from readme files. They should just provide succinct descriptions of the implementation state. This could be worded similar to:

"C# objects need not be manually disposed, as the memory management is handled by the _.NET Garbage Collector_ that calls the `IBaseObject::dispose()` method on object destruction. Notably, users still have the options of manually freeing openDAQ objects by calling the aforementioned `dispose()` method."